AsyncDisplayKit/Details/ASRangeDelegate.h

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <AsyncDisplayKit/ASLayoutRangeType.h>
+
+@class ASDisplayNode;
+
+@protocol ASRangeDelegate <NSObject>
```

Comment:
our implementations of ASRangeDelegate are ASPreloadRangeDelegate and ASRenderRangeDelegate.  renaming these to start with ASRangeDelegate would be clearer but clunkier.  maybe ASRangeObserver or ASRangeHandler?  then we'd have ASRangeHandlerPreload and ASRangeHandlerRender